### PR TITLE
A second batch of Pydantic models for rest/client/account.py

### PR DESCRIPTION
**Pull request by DMRobertson**
More of #13183.

`models.py` is getting a bit messy. It might be worth moving some of its contents into `rest/client/account.py`.

### CODE REVIEW

File: synapse/handlers/account.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def __init__(self, hs: "HomeServer"):
 
     async def get_account_statuses(
         self,
-        user_ids: List[str],
+        user_ids: Sequence[str],
```
Comment (clokep) [inline synapse/handlers/account.py:36]: I think we try not to do `Sequence[str]` since a `str` is also a `Sequence[str]`?

Comment (DMRobertson) [reply synapse/handlers/account.py:36]: Oh true. Context:

- I wanted to annotate the `user_ids` on the request body as immutable where possible.
- But I also wanted to use Pydantic's `conlist()` to enforce a minimum length---regrettably there doesn't seem to be a `contuple()`.

I'll drop the first bullet (which will be no worse than today).

File: synapse/rest/client/account.py [hunk 2]
```diff
@@ -494,16 +491,15 @@ async def on_GET(self, request: Request) -> None:
                 400, "Adding an email to your account is disabled on this server"
             )
 
-        sid = parse_string(request, "sid", required=True)
-        token = parse_string(request, "token", required=True)
-        client_secret = parse_string(request, "client_secret", required=True)
-        assert_valid_client_secret(client_secret)
+        body = parse_and_validate_json_object_from_request(
+            request, AddThreepidSubmitTokenBody
+        )
```
Comment (DMRobertson) [inline synapse/rest/client/account.py:496]: This is bogus: we're expecting query parameters here, not a request body.

File: synapse/rest/client/account.py [hunk 3]
```diff
@@ -400,23 +404,16 @@ def __init__(self, hs: "HomeServer"):
...
-        next_link = body.get("next_link")  # Optional param
-
-        msisdn = phone_number_to_msisdn(country, phone_number)
+        msisdn = phone_number_to_msisdn(body.country, body.phone_number)
 
         if not await check_3pid_allowed(self.hs, "msisdn", msisdn):
             raise SynapseError(
                 403,
+                # TODO: is this error message accurate? Looks like we've only rejected
+                #       this phone number, not necessarily all phone numbers
```
Comment (babolivier) [inline synapse/rest/client/account.py:416]: It's not accurate indeed. Looks like the previous error message raised in this case was mis-interpreted in https://github.com/matrix-org/synapse/pull/3789.

**Review by babolivier — APPROVED** [review]

File: synapse/rest/client/account.py (hunk 3 above)
Comment (DMRobertson) [reply synapse/rest/client/account.py:416]: I wondered if this was possibly intended as a spam deterrent.

**Pull request merged**